CONTRIBUTING.md

```diff
@@ -98,7 +93,42 @@ Right after opening project, Visual Studio will read cmake files and generate ca
 
 ### Linux / Mac
 
-VCPKG can also be used in Linux to avoid installing libraries globally. Follow instructions [here](#vcpkg) to use VCPKG in Linux.
+#### VCPKG
+
+VCPKG can be used to download packages sources, headers and build libraries for whatever TRIPLET is set up (platform/arq).
+VCPKG maintains any installed package inside its own folder, allowing to have multiple vcpkg folder with different dependencies installed on each. This is also great because you don't have to install dependencies globally on your system.
+
+Follow next steps to install VCPKG and have it linked to cmake
+
+```bash
+# Clone vcpgk:
+git clone https://github.com/Microsoft/vcpkg.git
+# (consider this path as PATH_TO_VCPKG)
+cd vcpkg
+# build vcpkg (remove .bat on Linux/Mac)
+./bootstrap-vcpkg.sh
+
+# Linux:
+./vcpkg install --triplet x64-linux curl cmocka
+export VCPKG_DEFAULT_TRIPLET=x64-linux
+export VCPKG_ROOT=PATH_TO_VCPKG #replace PATH_TO_VCPKG for where vcpkg is installed
+
+# Mac:
+./vcpkg install --triplet x64-osx curl cmocka
+export VCPKG_DEFAULT_TRIPLET=x64-osx
+export VCPKG_ROOT=PATH_TO_VCPKG #replace PATH_TO_VCPKG for where vcpkg is installed
+```
+
+> Note: On macOS, `.\bootstrap-vcpkg` may fail if your version of the C++ toolchain is not new enough to support vcpkg. To resolve
+this, vcpkg recommends installing `gcc@6` from Homebrew (`brew install gcc@6`), then re-run the bootstrapping script.
+
+#### Debian
+
+Alternatively, for Ubuntu 18.04 you can use:
+
+`sudo apt install build-essential cmake libcmocka-dev libcmocka0 gcovr lcov doxygen curl libcurl4-openssl-dev libssl-dev`
```

Comment:
Are all these required? I recently got the repo building on Ubuntu (using WSL) and I didn't need all these (that said, I only built/test the sources & docs, not samples). Did you find these to be needed in your experience? For example, isn't `build-essential` already there out of the box (including gcc/make)?

Reply:
Depends on the box and what you've installed. I don't consider WSL since it's designed with devs in mind.

In my experience Ubuntu Server LTS, Raspbian-Lite and a Debian I used on an iMX6 device did not have them. I copied these from IOT C-SDK: https://github.com/Azure/azure-iot-sdk-c/blob/master/build_all/linux/setup.sh.

What I'm not 100% sure of are the ones starting with `curl` since that's only required for certain types of builds.

Reply:
Also see other libs having the same requirements: https://github.com/eclipse/paho.mqtt.c#build-requirements--compilation-using-cmake

`apt-get install build-essential gcc make cmake cmake-gui cmake-curses-gui`